pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java

```diff
@@ -1490,6 +1494,87 @@ public FunctionState getFunctionState(final String tenant,
         return value;
     }
 
+    public void putFunctionState(final String tenant,
+                                 final String namespace,
+                                 final String functionName,
+                                 final String key,
+                                 final String stateJson,
+                                 final String clientRole,
+                                 final AuthenticationDataSource clientAuthenticationDataHttps) {
+
+        if (!isWorkerServiceAvailable()) {
+            throwUnavailableException();
+        }
+
+        try {
+            if (!isAuthorizedRole(tenant, namespace, clientRole, clientAuthenticationDataHttps)) {
+                log.error("{}/{}/{} Client [{}] is not admin and authorized to put state for {}", tenant, namespace,
+                        functionName, clientRole, ComponentTypeUtils.toString(componentType));
+                throw new RestException(Status.UNAUTHORIZED, "client is not authorize to perform operation");
+            }
+        } catch (PulsarAdminException e) {
+            log.error("{}/{}/{} Failed to authorize [{}]", tenant, namespace, functionName, e);
+            throw new RestException(Status.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+
+        if (null == worker().getStateStoreAdminClient()) {
+            throwStateStoreUnvailableResponse();
+        }
+
+        FunctionState state = new Gson().fromJson(stateJson, FunctionState.class);
```

Comment:
Please use ObjectMapperFactory.getThreadLocal() instead of creating anew Gson object

Reply:
done